File: src/providers/anthropic.ts
import type { Provider } from "../core/types.js";

export const anthropic: Provider = {
  name: "anthropic",
  displayName: "Anthropic",
  keyEnvVar: "ANTHROPIC_API_KEY",
  detect: (key) => /^sk-ant-[A-Za-z0-9_-]{16,}$/.test(key),
  validate: async (key, { http, signal }) => {
    const res = await http("https://api.anthropic.com/v1/models", {
      headers: {
        "x-api-key": key,
        "anthropic-version": "2023-06-01",
      },
      signal,
    });
    if (!res.ok) return { status: res.status, ok: false };
    const body = (await res.json().catch(() => ({}))) as { data?: unknown[] };
    return {
      status: res.status,
      ok: true,
      metadata: { modelCount: Array.isArray(body.data) ? body.data.length : 0 },
    };
  },
};

File: python/src/llm_key_validator/providers/anthropic.py
from __future__ import annotations

import re

from ..core.types import Provider, ProviderRawResult, ValidationContext

_DETECT = re.compile(r"^sk-ant-[A-Za-z0-9_-]{16,}$")


def _validate(key: str, ctx: ValidationContext) -> ProviderRawResult:
    res = ctx.http(
        "https://api.anthropic.com/v1/models",
        headers={"x-api-key": key, "anthropic-version": "2023-06-01"},
    )
    if not res.ok:
        return ProviderRawResult(status=res.status, ok=False)
    body = res.json() or {}
    data = body.get("data") if isinstance(body, dict) else None
    count = len(data) if isinstance(data, list) else 0
    return ProviderRawResult(status=res.status, ok=True, metadata={"modelCount": count})


anthropic = Provider(
    name="anthropic",
    display_name="Anthropic",
    key_env_var="ANTHROPIC_API_KEY",
    detect=lambda k: bool(_DETECT.match(k)),
    validate=_validate,
)
